doc/development/adding_operators.rst

```diff
@@ -209,6 +221,27 @@ FlipAndRotate(0.1, wires=['q3', 'q1'])
 >>> op.adjoint()
 FlipAndRotate(-0.1, wires=['q3', 'q1'])
 
+Once the class has been created, you can run a suite of validation checks using :func:`.ops.functions.assert_valid`.
+This function will warn you of some common errors in custom operators.
+>>> qml.ops.functions.assert_valid(op)
+
+If the above operator ommitted the ``_unflatten`` custom definition, it would raise:
+
+.. code-block::
+
+    File /pennylane/operation.py:1599, in Operator._unflatten(cls, data, metadata)
+      1575 """Recreate an operation from its serialized format.
+      1576 
+      1577 Args:
+      (...)
+      1596 
+      1597 """
+      1598 hyperparameters_dict = dict(metadata[1])
+    -> 1599 return cls(*data, wires=metadata[0], **hyperparameters_dict)
+
+    TypeError: FlipAndRotate.__init__() got an unexpected keyword argument 'wires'
```

Comment:
Is this line correct? I guess I expected something that mentions unflatten.

Reply:
I'll add a `try-except` around that check with a better error message.